Initial Commit

roiroooiii opened the pull request:
## Pull Request Description

Please replace this with a detailed description and motivation of the changes. 
You can tick the checkboxes below with an 'x' between square brackets or just check them after publishing the PR. 
If this PR contains a breaking change, list dependent PRs and try to push all related PRs at the same time.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [ ] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [ ] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [ ] I have complied with the [Submission Checklist](http://analogdevicesinc.github.io/no-OS/contributing.html#submission-checklist)
- [ ] I have performed a self-review of the changes
- [ ] I have commented my code, at least hard-to-understand parts
- [ ] I have build all projects affected by the changes in this PR
- [ ] I have tested in hardware affected projects, at the relevant boards
- [ ] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/analogdevicesinc/no-OS?pullRequest=2657) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/analogdevicesinc/no-OS?pullRequest=2657) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/analogdevicesinc/no-OS?pullRequest=2657) it.</sub>

cencarna commented on drivers/platform/maxim/max32690/maxim_rtc.h:
```diff
@@ -34,4 +34,24 @@
 #ifndef MAXIM_RTC_H_
 #define MAXIM_RTC_H_
 
+#define MSEC_TO_RSSA(x) \
```
Where is this used?

cencarna commented on projects/max32690_rtc_test/src/examples/basic_example/basic_example.c:
```diff
@@ -0,0 +1,187 @@
+}
+int basic_example_main()
+{
+	// Initialize the UART
```
ditto: comment formatting

cencarna commented on projects/max32690_rtc_test/src/examples/basic_example/basic_example.c:
```diff
@@ -0,0 +1,187 @@
+{
+	irq_flag = 1; // Set the flag to indicate that the RTC interrupt has occurred
+	no_os_rtc_set_cnt(rtc_desc, 0); // Reset the RTC count to 0
+	no_os_rtc_set_irq_time(rtc_desc, 10); // Reset the alarm to trigger again in 10 seconds
```
why reset rtc irq time again? is this erased on interrupt?

cencarna commented on projects/max32690_rtc_test/src/examples/basic_example/basic_example.c:
```diff
@@ -0,0 +1,187 @@
+
+void RTC_CB(void)
+{
+	irq_flag = 1; // Set the flag to indicate that the RTC interrupt has occurred
```
maybe wrap this around `no_os_irq_disable` and `no_os_irq_enable`. If you want this minimal, move rtc reset to the main loop after asserting `irq_flag`.

cencarna commented on projects/max32690_rtc_test/src/examples/basic_example/basic_example.c:
```diff
@@ -0,0 +1,187 @@
+		printf("RTC IRQ controller initialization failed.\n");
+		return ret;
+	}
+		ret = no_os_irq_enable(rtc_irq_desc, 3);
```
Fix indentation.

cencarna commented on projects/max32690_rtc_test/src/examples/basic_example/basic_example.c:
```diff
@@ -0,0 +1,187 @@
+	if (irq_flag){
+		no_os_gpio_set_value(led, NO_OS_GPIO_LOW); // Turn on the LED
+		no_os_mdelay(5000); // Delay for 5 seconds
+		ret = no_os_irq_register_callback(rtc_irq_desc, 3, &rtc_callback);
```
why re-register irq callback?

cencarna commented on projects/max32690_rtc_test/src/examples/basic_example/basic_example.c:
```diff
@@ -0,0 +1,187 @@
+			return ret;
+		}
+	}
+	else{
```
remove braces

cencarna commented on projects/max32690_rtc_test/src/examples/basic_example/basic_example.c:
```diff
@@ -0,0 +1,187 @@
+		// print time
+		ret = no_os_rtc_get_cnt(rtc_desc, &timer_cnt);
+		if (ret) 
+		{
```
inline brace

roiroooiii replied:
Fixed by running astyle. Thanks

cencarna reviewed (commented):
Other maxim MCUs also has RTC support. You may also want to add support for those MCUs here.

roiroooiii commented:
V1 -> V2

- Fixed Comments from //  to  /**/ Format
- Run astyle script to fix indentation issues and bracing
- Added appropriate whitespaces
- Changed printf -> pr_info
- Removed re-registration of callback function in main loop

roiroooiii commented on drivers/api/no_os_rtc.c:
```diff
@@ -0,0 +1,173 @@
+ * @param param - The structure that contains the RTC parameters.
+ * @return 0 in case of success, -ERROR otherwise.
+ */
+int32_t no_os_rtc_init(struct no_os_rtc_desc **device,
```
I retained usage of int32_t since the existing no_os_rtc.h in the main branch uses int32_t. I hope this is okay

kister-jimenez commented:
Can the project be a generic RTC example project with multiple platforms supported instead?

kister-jimenez commented:
/AzurePipelines run

CiprianRegus commented:
Hi @frmanabat,

Any update on this one, or can we close it for the moment?
